ui/admin/app/components/composed/ComboBox.tsx

```diff
@@ -94,16 +112,85 @@ function ComboBoxList<T extends BaseOption>({
     clearLabel,
     onChange,
     options,
-    placeholder = "Filter...",
     setOpen,
+    suggested,
     value,
+    placeholder = "Filter...",
+    emptyLabel = "No results found.",
 }: { setOpen: (open: boolean) => void } & ComboBoxProps<T>) {
-    const isGrouped = options.every((option) => "heading" in option);
+    const [filteredOptions, setFilteredOptions] =
+        useState<typeof options>(options);
+
+    const filterOptions = (
+        items: (T | GroupedOption<T>)[],
+        searchValue: string
+    ): (T | GroupedOption<T>)[] =>
+        items.reduce(
+            (acc, option) => {
+                const isSingleValueMatch =
+                    "name" in option &&
+                    option.name
+                        ?.toLowerCase()
+                        .includes(searchValue.toLowerCase());
+                const isGroupHeadingMatch =
+                    "heading" in option &&
+                    option.heading
+                        .toLowerCase()
+                        .includes(searchValue.toLowerCase());
+
+                if (isGroupHeadingMatch || isSingleValueMatch) {
+                    return [...acc, option];
+                }
+
+                if ("heading" in option) {
+                    const matches = filterOptions(option.value, searchValue);
+                    return matches.length > 0
+                        ? [
+                              ...acc,
+                              {
+                                  ...option,
+                                  value: matches,
+                              },
+                          ]
+                        : acc;
+                }
+
+                return acc;
+            },
+            [] as (T | GroupedOption<T>)[]
+        );
+
+    const sortBySuggested = (
+        a: T | GroupedOption<T>,
+        b: T | GroupedOption<T>
+    ) => {
+        // Handle nested groups - keep original order
+        if ("heading" in a || "heading" in b) return 0;
+
+        const aIsSuggested = a.name && suggested?.includes(a.name);
+        const bIsSuggested = b.name && suggested?.includes(b.name);
+
+        // If both or neither are suggested, maintain original order
+        if (aIsSuggested === bIsSuggested) return 0;
+        // Sort suggested items first
+        return aIsSuggested ? -1 : 1;
+    };
+
+    const handleValueChange = (value: string) => {
+        setFilteredOptions(filterOptions(options, value));
+    };
```

Comment:
same here. this should be handled by the parent. Especially if we ever need a combobox that displays data paginated by the server

Reply:
🙌 Good call out, moved the suggestions out of combobox! Thanks!